internal/coreinternal/attraction/type_converter.go

```diff
@@ -0,0 +1,57 @@
+package attraction
+
+import (
+	"strconv"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+func convertValue(to string, v pdata.AttributeValue) {
+	switch to {
+	case "string":
+		switch v.Type().String() {
+		case "STRING":
+		default:
+			v.SetStringVal(v.AsString())
+		}
+	case "int":
+		switch v.Type().String() {
```

Comment:
Same here
```suggestion
		switch v.Type() {
```

And using `pdata.AttributeValueTypeInt` and so forth